test/extensions/access_loggers/file/config_test.cc

```diff
@@ -73,6 +74,86 @@ TEST(FileAccessLogConfigTest, FileAccessLogTest) {
   EXPECT_NE(nullptr, dynamic_cast<FileAccessLog*>(instance.get()));
 }
 
+TEST(FileAccessLogConfigTest, FileAccessLogJsonTest) {
+  envoy::config::filter::accesslog::v2::AccessLog config;
+
+  envoy::config::accesslog::v2::FileAccessLog fal_config;
+  fal_config.set_path("/dev/null");
+
+  auto json_format = new ProtobufWkt::Struct;
```

Comment:
Probably don't want to be doing any raw `new` here, I think either make it stack allocated or a `make_unique`.

Reply:
I was having some trouble with that earlier, will give it another try

Reply:
So as far as I can tell, calling `set_allocated_*` with a non-null pointer causes protobuf to take ownership of that object, including releasing the object later. So I think using `new` is actually ok here.

Indeed, if I switch to using a `unique_ptr`, I get a crash, presumably because the object gets freed twice, once by protobuf and the other by c++. It does work fine if you use `make_unique` and then `release()` the pointer when passing it to `set_allocated`. I'm happy to do that if you want, although functionally, the effect is the same. Other option is to leave a comment on the call to `new` explaining why this is ok here.

Reply:
The usual pattern is to have protobuf do this for you by reordering and doing a `fal_config.mutable_allocated_json_format()` first.

Reply:
Figured this out. Thanks for the tip, that was the missing piece